detox/test/e2e/35.overlay.test.js

```diff
@@ -0,0 +1,46 @@
+const { expectToThrow } = require('./utils/custom-expects');
+
+describe(':ios: Overlay', () => {
+  let showOverlayButton;
+  let verticalScrollView;
+
+  beforeEach(async () => {
+    await device.reloadReactNative();
+    await device.launchApp({newInstance: true});
+
+    await element(by.text('Overlay')).tap();
+
+    showOverlayButton = await element(by.id('ShowOverlayButton'));
+    await expect(showOverlayButton).toBeVisible();
+
+    verticalScrollView = await element(by.id('VerticalScrollView'));
+  });
+
+  describe('button', () => {
+    it('should be visible when overlay is not shown', async () => {
+      await expect(verticalScrollView).toBeVisible();
+    });
+
+    it('should not be visible when overlay is not shown', async () => {
+      await showOverlayButton.tap();
+      await expect(verticalScrollView).not.toBeVisible();
+    });
+
+    it('should not be intractable when overlay is shown', async () => {
+      await showOverlayButton.tap();
+      await expectToThrow(() => showOverlayButton.tap());
+    });
+  });
+
+  describe('scroll view', () => {
+    it('should be intractable when overlay is not shown', async () => {
+      await verticalScrollView.scrollTo('bottom');
+      await expect(showOverlayButton).not.toBeVisible();
+    });
```

Comment:
I think is not necessary because it is tested in the actions suite

Reply:
Basically you are right but not really, since I'm testing the whole screen in this test I made sure that we start from the expected state (where the button disappears after scrolling). This covers the case where the scroll view might be not-scrollable (for example, too few items on some screen).

Reply:
Still kind of sounds like this: https://github.com/wix/Detox/blob/2c6478d6137bd0f3b3d3e92ee30fa82ad6f1d365/detox/test/e2e/03.actions-scroll.test.js#L33

Reply:
I disagree, I'm not testing the functionality of a scroll view, but the expectation from the initial state of this screen. Let's discuss about this f2f?